core/java/com/google/instrumentation/common/Timestamp.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.common;
+
+/**
+ * A representation of an instant in time. The instant is the number of nanoseconds after the number
+ * of seconds since the Unix Epoch.
+ */
+public class Timestamp {
+  /**
+   * Creates a new timestamp from given seconds and nanoseconds.
+   */
+  public static Timestamp create(long seconds, int nanos) {
```

Comment:
I think this should either normalize the inputs or throw an exception if nanos is out of range, since equals() compares the fields.